reana/reana_dev/git.py

```diff
@@ -1586,4 +1590,201 @@ def git_tag(component, exclude_components):  # noqa: D301
+
+    Example:
+    ```
+    ## [0.9.8](https://github.com/reanahub/reana-commons/compare/0.9.7...0.9.8) (2024-03-01)
+
+
+    ### Build
+
+    * **python:** change extra names to comply with PEP 685 [...]
+    ```
+
+    becomes
+
+    ```
+    ### reana-commons [0.9.8](https://github.com/reanahub/reana-commons/compare/0.9.7...0.9.8) (2024-03-01)
+
+    * [Build] **python:** change extra names to comply with PEP 685 [...]
+    ```
+    """
+    changelog_path = os.path.join(get_srcdir(component), "CHANGELOG.md")
+    with open(changelog_path) as f:
+        changelog_lines = f.readlines()
+
+    formatted_lines = []
+    is_version_to_add = False
+    current_section = ""
+    for line in changelog_lines:
+        # check if release in header is part of releases we are interested in
+        matches = re.match(r"##\s+\[?([\d.]+)", line)
+        if matches:
+            is_version_to_add = matches.group(1) in versions
+        if not is_version_to_add:
+            continue
+
+        if line.startswith("### "):
+            # commit type (e.g. fix, feat, ...)
+            current_section = line[len("### ") :].strip()
+        elif line.startswith("## "):
+            # release header
+            line = f"### {component}" + line[len("##") :]
+            if formatted_lines:
+                # add empty line before previous release changelog
+                formatted_lines.append("\n")
+            formatted_lines.append(line)
+            formatted_lines.append("\n")
+        elif line.startswith("*"):
+            # release please format, bullet points with '*'
+            formatted_lines.append(f"* [{current_section}]" + line[1:])
+        elif line.startswith("-"):
+            # old changelog format, bullet points with '-'
+            formatted_lines.append("*" + line[1:])
+
+    return formatted_lines
+
+
+def append_after_version_changelog(component, version, new_lines):
+    """Append the given lines after the changelog of the provided version."""
+    changelog_path = os.path.join(get_srcdir(component), "CHANGELOG.md")
+    with open(changelog_path) as changelog_file:
+        changelog_lines = changelog_file.readlines()
+
+    # find the idx of the release that follows the given one
+    idx_insert = None
+    found_given_version = False
+    for i, line in enumerate(changelog_lines):
+        if line.startswith("## "):
+            if version in line:
+                found_given_version = True
+            elif found_given_version:
+                idx_insert = i
+                break
+
+    if idx_insert is None:
+        raise ValueError(f"Could not find changelog of {component} {version}")
+
+    new_changelog = (
+        changelog_lines[:idx_insert] + new_lines + changelog_lines[idx_insert:]
+    )
+
+    with open(changelog_path, "w") as changelog_file:
+        changelog_file.writelines(new_changelog)
+
+
+@git_commands.command(name="git-aggregate-changelog")
+@click.option(
+    "--previous-reana-client",
```

Comment:
We could enrich the CLI later so that users could pass `-c` as to the other `reana-dev` commands. This might be useful to generate quickly an overview of all the changes from version X to version Y for example.

Such a CLI interface could look something like:

```
-c r-client -- since-version 0.9.2
-c r-client --since-date=2023-01-01
```

Or, by analogy with `git-checkout-pr`, we should probably have a tuple of values to allow assembling several components:

```
-b r-client 0.9.1 -b r-j-controller 0.9.0 
```

Just some wild musings for later, not for now.  

That said, if we do decide to go for this one day, to allow aggregating news from more than one release, then the CLI option name could change, so perhaps we could forward-name the option to be something like `--since-reana-client-version 0.9.2` instead of `--previous-reana-client`? :thinking: